### ISSUE TITLE
Timepicker not cleared on form reset with half value

### ISSUE THREAD

**Issue by DGolverdingen** [telerik/kendo-angular#1398]
## I'm submitting a..

* Bug report 

## Current behavior
When deleting half the time. (only the hour or the minute), on form reset the other value with not be cleared. For example it keeps showing "10:minute"

## Expected behavior
On form reset every time picker should display default "hour:minute"

## Minimal reproduction of the problem with instructions
https://plnkr.co/edit/U6aNrLHFCifu6lG6mBwe?p=preview

## What is the motivation or use case for changing the behavior?
It looks ugly on a form that not all values are completely cleared.

**Comment by ggkrustev** [comment]
Hi @DGolverdingen,
Indeed, the component doesn't explicitly reset the input value when it is partially typed. Basically, the component considers the half-typed value as `null` and setting it again to `null` (form.reset()) is not considered as change.

I do agree that the input value should be reset too, thus improving overall UX. We will plan this enhancement in some of our future releases.

**Comment by ggkrustev** [comment]
Our investigation proved that the Angular form won't call component writeValue, because the value hasn't changed (it is still `null`). The solution that we come up with is to allow the usage of the component's `writeValue` to reset the input manually. Basically, the developer should call the DatePicker/TimePicker writeValue method with `null` to manually reset input value.

```
<button (click)="dp.writeValue(null)">reset</button>
<kendo-datepicker #dp formControlName="start"></kendo-datepicker>
```

Unfortunately, there is no other feasible option on our side to handle the discussed situation. The best will be the one that develops the form to reset the components values manually, as they know when and why to reset the inputs.

The enhancement will be available soon in the DEV channel of the @progress/kendo-angular-dateinputs package.

**Comment by ggkrustev** [comment]
A fix is released in `v3.4.0-dev.201805210833`. The easiest to try the change is to install the `DEV` release:
`npm i @progress/kendo-angular-dateinputs@dev`

**Comment by tsvetomir** [comment]
Fixed in v3.4.0
